File: 409-longest-palindrome/longest-palindrome.java
import java.util.*;

class Solution {
    public int longestPalindrome(String s) {
        Map<Character, Integer> map = new HashMap<>();
        for (char c : s.toCharArray()) {
            map.put(c, map.getOrDefault(c, 0) + 1);
        }

        int len = 0;
        boolean oddFound = false;
        for (int value : map.values()) {
            if (value % 2 == 0) {
                len += value;
            } else {
                len += value - 1;  // use the even part
                oddFound = true;   // mark that odd count exists
            }
        }

        if (oddFound) {
            len += 1;  // add one odd character in the middle
        }

        return len;
    }
}

File: 409-longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        map={}
        for c in s:
            map[c]=map.get(c,0)+1

        even=0
        odd=0
        c=0
        for k,v in map.items():
            if v%2==0:
                even+=v
            else:
                odd+=v  
                c+=1
        
        if odd>0:
            return even+odd-c+1
        else:
            return even
